src/ring_theory/simple_module.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors : Aaron Anderson
+-/
+
+import linear_algebra.basic
+
+/-!
+# Simple Modules
+
+## Main Definitions
+  * `is_simple_module` indicates that a module has no proper submodules
+  (the only submodules are `⊥` and `⊤`).
+  * A `division_ring` structure on the endomorphism ring of a simple module.
+
+## Main Results
+  * Schur's Lemma: `bijective_or_eq_zero_of_simple` shows that a linear map between simple modules
+  is either bijective or 0, leading to a `division_ring` structure on the endomorphism ring.
+
+## TODO
+  * Semisimple modules, Artin-Wedderburn Theory
+  * Unify with the work on Schur's Lemma in a category theory context
+
+-/
+
+variables (R : Type*) [comm_ring R] (M : Type*) [add_comm_group M] [module R M]
+
+/-- A module is simple when it has only two submodules, `⊥` and `⊤`. -/
+abbreviation is_simple_module := (is_simple_lattice (submodule R M))
+
+-- Making this an instance causes the linter to complain of "dangerous instances"
+theorem is_simple_module.nontrivial [is_simple_module R M] : nontrivial M :=
+⟨⟨0, begin
+  have h : (⊥ : submodule R M) ≠ ⊤ := bot_ne_top,
+  contrapose! h,
+  ext,
+  simp [submodule.mem_bot,submodule.mem_top, h x],
+end⟩⟩
+
+variables {R} {M}  {N : Type*} [add_comm_group N] [module R N]
+
+namespace linear_map
+
+theorem injective_or_eq_zero_of_simple [is_simple_module R M] (f : M →ₗ[R] N) :
+  function.injective f ∨ f = 0 :=
+begin
+  rw [← ker_eq_bot, ← ker_eq_top],
+  apply eq_bot_or_eq_top,
+end
+
+theorem surjective_or_eq_zero_of_simple [is_simple_module R N] (f : M →ₗ[R] N) :
+  function.surjective f ∨ f = 0 :=
+begin
+  rw [← range_eq_top, ← range_le_bot_iff, le_bot_iff, or_comm],
+  apply eq_bot_or_eq_top,
+end
+
+/-- Schur's Lemma for linear maps between (possibly distinct) simple modules -/
+theorem bijective_or_eq_zero_of_simple [is_simple_module R M] [is_simple_module R N]
+  (f : M →ₗ[R] N) :
+  function.bijective f ∨ f = 0 :=
+begin
+  by_cases h : f = 0,
+  { right,
+    exact h },
+  exact or.intro_left _ ⟨or.resolve_right f.injective_or_eq_zero_of_simple h,
+    or.resolve_right f.surjective_or_eq_zero_of_simple h⟩,
+end
```

Comment:
would you mind adding some convenience lemmas like `injective_of_ne_zero` and such?

Reply:
Added!